### PR TITLE
New list_comparison_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #884

No hits on {lintr}

### CODE REVIEW

**Comment by AshesITR** [comment]
Add `mapplay()`?

File: R/list_comparison_linter.R [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#'   text = "unlist(lapply(x, sum)) > 10",
+#'   linters = list_comparison_linter()
+#' )
+#'
+#' @evalRd rd_tags("list_comparison_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+list_comparison_linter <- function() {
+  # TODO(michaelchirico): extend to cases where using simplify=FALSE implies a
+  #   list output, e.g. with sapply, replicate, mapply.
```
Comment (MichaelChirico) [inline R/list_comparison_linter.R:25]: @AshesITR  it's under the TODO -- `mapply()` may or may not return a list. e.g. ``mapply(`+`, 1:10, 1:10)``.

Comment (AshesITR) [reply R/list_comparison_linter.R:25]: Does `.mapply()` behave differently?

Comment (MichaelChirico) [reply R/list_comparison_linter.R:25]: `.mapply()` always returns list, see `mapply()` implementation which basically calls `simlify2array()` on `.mapply()` output:

https://github.com/r-devel/r-svn/blob/9f3b4bea25f93be3ae951e40f36bcf8e54a7201f/src/library/base/R/mapply.R#L19-L38

I'll add it

File: tests/testthat/test-list_comparison_linter.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  expect_lint("sapply(x, sum) > 10", NULL, list_comparison_linter())
+})
+
+local({
+  linter <- list_comparison_linter()
+  lint_msg <- rex::rex("a list(), is being coerced for comparison")
+
+
+  cases <- expand.grid(
+    list_mapper = c("lapply", "map", "Map"),
```
Comment (AshesITR) [inline tests/testthat/test-list_comparison_linter.R:11]: Add `.mapply` here

File: tests/testthat/test-list_comparison_linter.R [hunk 3]
```diff
@@ -0,0 +1,20 @@
+test_that("list_comparison_linter skips allowed usages", {
+  expect_lint("sapply(x, sum) > 10", NULL, list_comparison_linter())
+})
+
+local({
+  linter <- list_comparison_linter()
+  lint_msg <- rex::rex("a list(), is being coerced for comparison")
+
+
```
Comment (AshesITR) [inline tests/testthat/test-list_comparison_linter.R:9]: unnecessary newline

**Review by AshesITR — COMMENTED** [review]
Metadata and vectorization tests missing; the rest LGTM now 😊

**Review by AshesITR — APPROVED** [review]

**Pull request merged**